TitanRobotics2024/src/main/java/frc/robot/Subsystem/ClimberSubsystem.java

```diff
@@ -0,0 +1,177 @@
+    private double rotationTarget;
+    private double holdPosition = 0.0;
+    private double climberVelocity = 0.0;
+    private double climberPower = 0.0;
+    private double currentDistance = 0.0;
+    private double currentVelocity = 0.0;
+    private String name = "Climber"; // Replace with your name
+
+    // Singleton instances for left and right climbers
+    private static ClimberSubsystem leftInstance = null;
+    private static ClimberSubsystem rightInstance = null;
+
+    // Motors and encoders for left and right climbers
+    private ModifiedMotors motor;
+    private ModifiedEncoders encoder;
+    private SmartDashboardSubsystem smartDashboardSubsystem;
+
+    // Other properties and methods...
+    // Private constructor for initializing motors and encoders
+    private ClimberSubsystem(ModifiedMotors motor, ModifiedEncoders encoder, String name)
+    {
+
+        this.motor = motor;
+        this.encoder = encoder;
+        this.name = name;
+        smartDashboardSubsystem = SmartDashboardSubsystem.getInstance();
+        // Add initialization logic here
+    }
+
+    // Get instance for left climber
+    public static ClimberSubsystem getLeftInstance()
+    {
+        if (leftInstance == null)
+        {
+            leftInstance = new ClimberSubsystem(new ModifiedMotors(PortMap.CLIMBERMOTORLEFT.portNumber, "CANSparkMax"),
+                            new ModifiedEncoders(PortMap.CLIMBERLEFTENCODER_A.portNumber, PortMap.CLIMBERLEFTENCODER_B.portNumber, "QuadratureEncoder"), "Left Climber");
+        }
+        return leftInstance;
+    }
+
+    // Get instance for right climber
+    public static ClimberSubsystem getRightInstance()
+    {
+        if (rightInstance == null)
+        {
+            rightInstance = new ClimberSubsystem(new ModifiedMotors(PortMap.CLIMBERMOTORRIGHT.portNumber, "CANSparkMax"),
+                            new ModifiedEncoders(PortMap.CLIMBERRIGHTENCODER_A.portNumber, PortMap.CLIMBERRIGHTENCODER_B.portNumber, "QuadratureEncoder"), "Right Climber");
+
+        }
+        return rightInstance;
+    }
+
+    // Set manual control for climber velocity
+    public void manualControl(double velocity)
+    {
+        this.climberVelocity = velocity;
+        this.climberState = "MANUAL";
+    }
+
+    // Stop the climber
+    public void stop()
+    {
+        this.climberState = "STOPPED";
+    }
+
+    // Set climber state to top
+    public void top()
+    {
+        this.climberState = "TOP";
+    }
+
+    // Set climber state to bottom
+    public void bottom()
+    {
+        this.climberState = "BOTTOM";
+    }
+
+    // Set climber state to hold
+    public void hold()
+    {
+        this.climberState = "HOLD";
+        this.holdPosition = currentDistance;
+    }
+
+    private void processState()
+    {
+        switch (climberState)
+        {
+            case "MANUAL":
+                //if (!((climberVelocity > 0.0 && atTop()) || (climberVelocity < 0.0 && atBottom()))) 
```

Comment:
Remove commented out code

Reply:
elise said keep for later

Reply:
You aren't using it, remove it. You don't want code that you haven't tested in your main code base leading to confusion on if it works or not